File: app/lib/api/reports_api.dart
import 'api_client.dart';

class ReportsApi {
  ReportsApi(this._client);
  final ApiClient _client;

  /// POST /api/reports — requires session.
  Future<Map<String, dynamic>> submit({
    required String targetKind,
    required String targetId,
    required String reason,
    String details = '',
    String? pageContext,
    String? label,
  }) async {
    return _client.requestJson(
      '/api/reports',
      method: 'POST',
      body: {
        'targetKind': targetKind,
        'targetId': targetId,
        'reason': reason,
        'details': details,
        if (pageContext != null && pageContext.isNotEmpty) 'pageContext': pageContext,
        if (label != null && label.isNotEmpty) 'label': label,
      },
    );
  }
}

File: src/utils/reportsApi.js
import { apiFetch } from '../auth/AuthContext';

function apiError(res, data) {
    const msg = (data && data.error) || `HTTP ${res.status}`;
    const err = new Error(msg);
    err.status = res.status;
    err.code = data && data.error;
    return err;
}

/** POST /api/reports — requires session. */
export async function submitReport(payload) {
    const res = await apiFetch('/api/reports', {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({
            targetKind: payload.targetKind,
            targetId: payload.targetId,
            reason: payload.reason,
            details: payload.details,
            pageContext: payload.pageContext,
            label: payload.label,
        }),
    });
    const data = await res.json().catch(() => ({}));
    if (!res.ok) throw apiError(res, data);
    return data;
}
